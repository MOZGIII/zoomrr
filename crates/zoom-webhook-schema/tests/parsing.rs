use zoom_webhook_schema::*;

#[test]
fn webhook() {
    let sample = include_bytes!("../../../testdata/samples/recording.completed.json");
    let value: Webhook = serde_json::from_slice(sample).unwrap();
    insta::assert_yaml_snapshot!(value);
}

#[test]
fn recording_completed() {
    let sample = include_bytes!("../../../testdata/samples/recording.completed.json");
    let value: Root<recording::completed::Payload> = serde_json::from_slice(sample).unwrap();
    insta::assert_yaml_snapshot!(value);
}

#[test]
fn endpoint_url_validation() {
    let sample = include_bytes!("../../../testdata/samples/endpoint.url_validation.json");
    let value: Root<endpoint::url_validation::Payload> = serde_json::from_slice(sample).unwrap();
    insta::assert_yaml_snapshot!(value);
}
