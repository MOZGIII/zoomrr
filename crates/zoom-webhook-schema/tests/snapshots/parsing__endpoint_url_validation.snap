---
source: crates/zoom-webhook-schema/tests/parsing.rs
assertion_line: 21
expression: value
---
event: endpoint.url_validation
event_ts: 1654503849680
payload:
  plainToken: qgg8vlvZRS6UYooatFL8Aw

